lib/route-utils/authPreHandlers.ts

```diff
@@ -0,0 +1,29 @@
+import { AuthFailedError } from '@lokalise/node-core'
+import type { FastifyReply, FastifyRequest, HookHandlerDoneFunction } from 'fastify'
+
+const BEARER_PREFIX = 'Bearer '
+const BEARER_PREFIX_LENGTH = BEARER_PREFIX.length
+
+export function createStaticTokenAuthPreHandler(configuredSecretToken: string) {
+  return function preHandlerStaticTokenAuth(
+    req: FastifyRequest,
+    _: FastifyReply,
+    done: HookHandlerDoneFunction,
+  ) {
+    const logger = req.reqContext?.logger
```

Comment:
@kibertoad What is the better way to ensure `reqContext` is present? `FastifyRequest` is extended by request context plugin. Shall we have dedicated type e.g. `FastifyRequest & { reqContext }`?

Reply:
sure, that sounds reasonable.

Reply:
OK, let me check this option.

Reply:
I'm dropping this idea. Code will be odd. Inead I'm introducing another param `loggerProvider` it's up to the client define the way of getting logger.

The problem is that requestContextPlugin needs to take `FastifyRequest` and produce `FastifyRequestWithContext` request object. So the preHandler would need to somehow dependent on usage of plugin, it's not possible since registration of plugin and prehandler happens in runtime.

Reply:
I agree, this is a better and more flexible approach